ts/components/icon/SessionNotificationCount.tsx

```diff
@@ -21,34 +20,48 @@ const StyledCountContainer = styled.div<{ shouldRender: boolean }>`
   font-weight: 700;
   background: var(--unread-messages-alert-background-color);
   transition: var(--default-duration);
-  opacity: ${props => (props.shouldRender ? 1 : 0)};
   text-align: center;
   color: var(--unread-messages-alert-text-color);
+  white-space: ${props => (props.centeredOnTop ? 'nowrap' : 'normal')};
 `;
 
-const StyledCount = styled.div`
+const StyledCount = styled.div<{ centeredOnTop: boolean }>`
   position: relative;
-  font-size: 0.6rem;
+  font-size: ${props => (props.centeredOnTop ? 'var(--font-size-xs)' : '0.6rem')};
 `;
 
-export const SessionNotificationCount = (props: Props) => {
-  const { count } = props;
-  const overflow = Boolean(count && count > 99);
-  const shouldRender = Boolean(count && count > 0);
+const OverflowingAt = (props: { overflowingAt: number }) => {
+  return (
+    <>
+      {props.overflowingAt}
+      <span>+</span>
+    </>
+  );
+};
 
-  if (overflow) {
-    return (
-      <StyledCountContainer shouldRender={shouldRender}>
-        <StyledCount>
-          {99}
-          <span>+</span>
-        </StyledCount>
-      </StyledCountContainer>
-    );
+const NotificationOrUnreadCount = ({
+  centeredOnTop,
+  overflowingAt,
+  count,
+}: Props & { overflowingAt: number; centeredOnTop: boolean }) => {
```

Comment:
overflowingAt and centeredOnTop can go into the Props type

Reply:
well some calls don't need it so I refactored it a bit